src/config.zig

```diff
@@ -177,6 +171,22 @@ pub const io_depth_read = 8;
 /// The maximum number of concurrent write I/O operations to allow at once.
 pub const io_depth_write = 8;
 
+// There must be enough messages to ensure that the replica can always progress,
+// i.e. it never deadlocks because it doesn't have a free message.
+const client_table_messages_max = clients_max;
+const journal_messages_max = io_depth_read + io_depth_write;
+const loopback_queue_messages_max = 1;
+const pipelining_messages_max = pipelining_max * (1 + replicas_max);
+const quorum_messages_max = 3 * replicas_max;
+
+/// The number of full-sized messages allocated at initialization by the message bus.
+pub const message_bus_messages_max = pipelining_messages_max + quorum_messages_max +
+    loopback_queue_messages_max + client_table_messages_max + journal_messages_max;
+
+/// The number of header-sized messages allocated at initialization by the message bus.
+/// These are much smaller/cheaper and we can therefore have many of them.
+pub const message_bus_headers_max = connections_max * connection_send_queue_max * 2;
```

Comment:
Could we tighten this a little more by looking for all the places where we use header-only messages?

At present, the value I tuned here is more of a thumbsuck. It's very nice now that we're starting to work out the exact allocation we need.